### ISSUE TITLE
Error and solution when calling self.model.evaluate in trainer.py line 500

### ISSUE THREAD

**Issue by lfdharo** [flairNLP/flair#2354]
Line 500 in trainer.py should be changed from:
                  train_part_eval_result, train_part_eval_loss = self.model.evaluate(
to

                    train_part_eval_result = self.model.evaluate(
                        train_part,
                        gold_label_type=self.model.label_type,
                        mini_batch_size=mini_batch_chunk_size,
                        num_workers=num_workers,
                        embedding_storage_mode=embeddings_storage_mode,
                        main_evaluation_metric=main_evaluation_metric
                    )

then, for the rest of things to work they should also be changed to:

                    result_line += (
                        f"\t{train_part_eval_result.loss}\t{train_part_eval_result.log_line}"
                    )
                    log.info(
                        f"TRAIN_SPLIT : loss {train_part_eval_result.loss} - {main_evaluation_metric[1]} ({main_evaluation_metric[0]}) {round(train_part_eval_result.main_score, 4)}"
                    )
